tee-worker/app-libs/stf/src/trusted_call_litentry.rs

```diff
@@ -19,186 +19,111 @@ extern crate sgx_tstd as std;
 
 use super::*;
 use crate::{
-	helpers::{ensure_enclave_signer_account, generate_challenge_code},
-	is_root, AccountId, IdentityManagement, MetadataOf, Runtime, StfError, StfResult,
+	helpers::{ensure_enclave_signer_account, generate_challenge_code, is_authorised_signer},
+	AccountId, IdentityManagement, MetadataOf, Runtime, StfError, StfResult,
 };
 use frame_support::{dispatch::UnfilteredDispatchable, ensure};
 use ita_sgx_runtime::RuntimeOrigin;
 use itp_stf_primitives::types::ShardIdentifier;
-use itp_utils::stringify::account_id_to_string;
 use lc_stf_task_sender::{
 	stf_task_sender::{SendStfRequest, StfRequestSender},
-	AssertionBuildRequest, IdentityVerificationRequest, RequestType, SetUserShieldingKeyRequest,
+	AssertionBuildRequest, IdentityVerificationRequest, RequestType,
 };
 use litentry_primitives::{
 	Assertion, ChallengeCode, ErrorDetail, Identity, ParentchainBlockNumber, UserShieldingKeyType,
 	ValidationData,
 };
 use log::*;
-use std::{format, sync::Arc, vec::Vec};
-
-type IMTCall = ita_sgx_runtime::IdentityManagementCall<Runtime>;
+use std::{format, vec::Vec};
 
 impl TrustedCallSigned {
-	pub fn set_user_shielding_key_preflight(
-		root: AccountId,
-		shard: &ShardIdentifier,
-		who: AccountId,
-		key: UserShieldingKeyType,
-		hash: H256,
-	) -> StfResult<()> {
-		ensure!(is_root::<Runtime, AccountId>(&root), StfError::MissingPrivileges(root));
-		let request = SetUserShieldingKeyRequest { shard: *shard, who, key, hash }.into();
-		let sender = StfRequestSender::new();
-		sender
-			.send_stf_request(request)
-			.map_err(|_| StfError::SetUserShieldingKeyFailed(ErrorDetail::SendStfRequestFailed))
-	}
-
-	pub fn set_user_shielding_key_runtime<NodeMetadataRepository>(
-		node_metadata_repo: Arc<NodeMetadataRepository>,
-		calls: &mut Vec<OpaqueCall>,
+	pub fn set_user_shielding_key_internal(
```

Comment:
These codes come from upstream, and I think it's best to provide a PR to upstream for modification if possible, otherwise the code for each merge of upstream may become more cumbersome.

Reply:
Thanks for the suggestion, yes we probably aren't going to change it as we want fewer deviations from upstream codebase when possible.